packages/api/models/rules.js

```diff
@@ -276,9 +253,9 @@ class Rule extends Manager {
     const deleteEventPromises = this.kinesisSourceEvents.map(
       (lambda) => this.deleteKinesisEventSource(item, lambda.eventType)
     );
-    const eventDelete = await Promise.all(deleteEventPromises);
-    item.rule.arn = eventDelete[0];
-    item.rule.logEventArn = eventDelete[1];
+    await Promise.all(deleteEventPromises);
+    item.rule.arn = null;
+    item.rule.logEventArn = null;
```

Comment:
Is this something where you may want to `delete item.rule.arn` (and `logEventArn`) instead of setting it to null? Though I'm not sure if that would make its way back to the database correctly.

Reply:
I'm not sure it matters here. This doesn't get written back to the db, since the whole rule will get deleted. I'm not sure we need to set it to anything here, but maybe null or delete just to be safe?

I looked and if we shouldn't have a rule of type kinesis with no event source. I'm fine to do whatever here though.

Reply:
I don't see any reason that we're actually returning item here.  It isn't being used.

Reply:
True. The assignments are changing the "original" rule object as a side effect, though. That also doesn't really matter because they are about to be replaced.. The more I dig into this rabbit hole the less sure I am `base.update()` captures everything. Is it really performing a deep update?